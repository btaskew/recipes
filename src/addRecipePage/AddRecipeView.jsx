import React from 'react';
import PropTypes from 'prop-types';

import {Alert} from 'react-bootstrap';
import {Redirect} from 'react-router';
import Page from 'components/Page';
import InputForm from 'components/InputForm';

function AddRecipeView(props) {
    return (
        <Page heading="Add recipe" loading={props.loading}>
            <InputForm
                submitForm={props.handleSubmit}
                buttonText="Add"
                inputText="Enter recipe name"
            />

            {props.error && <Alert bsStyle="danger">{props.errorMessage}</Alert>}

            {props.success && <Redirect to={props.redirectPath} />}
        </Page>
    );
}

AddRecipeView.propTypes = {
    error: PropTypes.bool.isRequired,
    handleSubmit: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired,
    success: PropTypes.bool.isRequired,
    errorMessage: PropTypes.string,
    redirectPath: PropTypes.string
};
AddRecipeView.defaultProps = {
    errorMessage: 'Error adding recipe',
    redirectPath: '/'
};

export default AddRecipeView;
