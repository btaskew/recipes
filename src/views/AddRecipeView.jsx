import React from 'react';

import {Alert} from 'react-bootstrap';
import {Redirect} from 'react-router';
import Page from 'components/utils/Page';
import InputForm from 'components/utils/InputForm';

function AddRecipeView(props) {
    return (
        <Page heading="Add recipe">
            <InputForm
                submitForm={props.handleSubmit}
                buttonText="Add"
                inputText="Enter recipe name"
            />

            {props.success && <Redirect to={props.redirectPath} />}

            {props.error && (
                <Alert className="topPadding" bsStyle="danger">
                    {props.errorMessage}
                </Alert>
            )}
        </Page>
    );
}

export default AddRecipeView;
