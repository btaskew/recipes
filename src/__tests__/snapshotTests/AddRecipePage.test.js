import React from 'react';
import renderer from 'react-test-renderer';

import AddRecipeView from 'addRecipePage/AddRecipeView';
import {MemoryRouter} from 'react-router';

describe('Add recipe page', () => {
    it('renders loading correctly', () => {
        const addRecipePage = renderer
            .create(
                <MemoryRouter>
                    <AddRecipeView
                        loading={true}
                        error={false}
                        success={false}
                        handleSubmit={() => {}}
                    />
                </MemoryRouter>
            )
            .toJSON();
        expect(addRecipePage).toMatchSnapshot();
    });

    it('renders an error correctly', () => {
        const addRecipePage = renderer
            .create(
                <MemoryRouter>
                    <AddRecipeView
                        loading={false}
                        error={true}
                        errorMessage={"Error"}
                        success={false}
                        handleSubmit={() => {}}
                    />
                </MemoryRouter>
            )
            .toJSON();
        expect(addRecipePage).toMatchSnapshot();
    });

    it('renders normally correctly', () => {
        const addRecipePage = renderer
            .create(
                <MemoryRouter>
                    <AddRecipeView
                        loading={false}
                        error={false}
                        success={false}
                        handleSubmit={() => {}}
                    />
                </MemoryRouter>
            )
            .toJSON();
        expect(addRecipePage).toMatchSnapshot();
    });
});
