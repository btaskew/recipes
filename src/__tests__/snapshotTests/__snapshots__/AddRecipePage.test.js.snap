// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add recipe page renders an error correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Add recipe
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="flex"
    >
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Enter recipe name"
        type="text"
        value=""
      />
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Add
      </button>
    </div>
    <div
      className="alert alert-danger"
      role="alert"
    >
      Error
    </div>
  </div>,
]
`;

exports[`Add recipe page renders loading correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Add recipe
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="alert alert-info"
      role="alert"
    >
      Loading...
    </div>
  </div>,
]
`;

exports[`Add recipe page renders normally correctly 1`] = `
Array [
  <div
    className="heading modal-header"
  >
    <h4
      className="modal-title"
    >
      Add recipe
      <span
        className="float-right active glyphicon glyphicon-home"
        href="/"
        onClick={[Function]}
        style={Object {}}
      />
    </h4>
  </div>,
  <div
    className="modal-body"
  >
    <div
      className="flex"
    >
      <input
        className="form-control"
        id={undefined}
        onChange={[Function]}
        placeholder="Enter recipe name"
        type="text"
        value=""
      />
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Add
      </button>
    </div>
  </div>,
]
`;
